### DISCUSSION TITLE
gpt-4-0613 - function calling

### DISCUSSION THREAD

**Discussion by stanislavlysenko0912** [n3d1117/chatgpt-telegram-bot#343] [Ideas]
After the release of gpt-4-0613 which allows you to run functions, I think it could be a good way to extend the functionality of the bot.

By making some kind of handy plugin mechanism for it. 
That people could offer their own ready-made-small solutions that might be useful.
Also a handy mechanism to turn them on/off.

@n3d1117  are you going to do something for this? 
Perhaps someone else can help with this, does anyone have any ideas for implementation?) 

I'm just not sure I can make an accessible mechanism to work for all of this 🥲

**Comment by n3d1117** [comment]
Already in the works :) stay tuned!

**Comment by n3d1117** [comment]
Initial implementation and discussion is up! https://github.com/n3d1117/chatgpt-telegram-bot/discussions/345
